src/main/webapp/assets/js/feedScript.js

```diff
@@ -605,16 +614,204 @@ function closeModal() {
+  invalidIds.forEach((id) => {
+    const elt = formElements.namedItem(id);
+    if (elt) {
+      elt.style.background = '#ff999966';
     }
+  });
+  if (errorMessages.length > 0) {
+    const modalError = document.createElement('div');
+    modalError.setAttribute('id', 'modal-input-error');
+    modalError.setAttribute('tabindex', '0');
+    modalError.innerHTML += '<ul>';
+    errorMessages.forEach((message) => {
+      modalError.innerHTML += '<li>' + message + '</li>';
+    });
+    modalError.innerHTML += '</ul>';
+    modalForm.insertBefore(modalError, submitModalButton);
   }
+}
+
+/**
+ * Checks if the month and day are a valid date.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateModalDate(invalidIds, errorMessages, formElements) {
+  const month = formElements.namedItem('modal-month').value;
+  const day = formElements.namedItem('modal-day').value;
+  const monthDayLengths = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+  if (month > 12 || month < 1) {
+    invalidIds.push('modal-month');
+    errorMessages.push('month must be between 1 and 12');
+  }
+  if (day < 1 || day > monthDayLengths[month - 1] || isBlank(day)) {
+    invalidIds.push('modal-day');
+    errorMessages.push('invalid date');
+  }
+}
+
+/**
+ * Checks if the start and end time are valid.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateModalTime(invalidIds, errorMessages, formElements) {
+  const startHour = parseInt(formElements.namedItem('modal-start-hour').value, 10);
+  const startMinute = parseInt(formElements.namedItem('modal-start-minute').value, 10);
+  const startAMorPM = parseInt(formElements.namedItem('start-am-or-pm').value, 10);
+  const endHour = parseInt(formElements.namedItem('modal-end-hour').value, 10);
+  const endMinute = parseInt(formElements.namedItem('modal-end-minute').value, 10);
+  const endAMorPM = parseInt(formElements.namedItem('end-am-or-pm').value, 10);
+
+  // Check if the hours fall between 1-12.
+  if (startHour < 1 || startHour > 12 || isNaN(startHour)) {
+    invalidIds.push('modal-start-hour');
+    errorMessages.push('start hour must be between 1 - 12');
+  }
+  if (endHour < 1 || endHour > 12 || isNaN(endHour)) {
+    invalidIds.push('modal-end-hour');
+    errorMessages.push('end hour must be between 1 - 12');
+  }
+  // Check if the minutes fall between 0 - 60.
+  if (startMinute < 0 || startMinute >= 60 || isNaN(startMinute)) {
+    invalidIds.push('modal-start-minute');
+    errorMessages.push('start minute must be between 00 - 59');
+  }
+  if (endMinute < 0 || endMinute >= 60 || isNaN(startMinute)) {
+    invalidIds.push('modal-end-minute');
+    errorMessages.push('end minute must be between 00 - 59');
+  }
+  // Check if the end time is after the start time.
+  if ((startAMorPM === 'pm' && endAMorPM === 'am') ||
```

Comment:
To improve readability I would recommend extracting some of these into variables and doing the or-check. Naming the variables would probably be the hardest part.

Reply:
Great idea! I'll make the change.